### PR TITLE
PyBoy boot ROM

### PR DESCRIPTION
**Pull request by Baekalfen**
Solves #89.

The logo it shows on boot is pretty rough, but can easily be changed with the provided scripts and Makefile.

It also initialize the window classes with a white background, as the GameBoy would. Otherwise, the boot logo will be on a black background.

### CODE REVIEW

**Comment by krs013** [comment]
`bootrom.bin` does not seem to be a part of this PR, but I think you meant to put it in.

Also, even though the default boot-ROM is now our replacement, the warning message still prints out if the `DMG_ROM.bin` rom isn't found (even though it's not being used). That message should probably just be removed at this point.

File: .flake8 [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-# E128: flake8 is either buggy, or the rule is nonsensical: Continuation line under-indented for visual indent
-# E221: Sometimes alignment of operators is better. Use common sense: "multiple spaces before operator"
-# E203: Sometimes alignment of commas is better. Use common sense: "whitespace before ','"
-# W503: Sometimes common sense prevail: "line break before binary operator"
-# W504: Sometimes common sense prevail: "line break after binary operator"
-# E226: "If operators with different priorities are used, __consider__ adding whitespace around the operators with the lowest priority(ies). Use your own judgment;..."
-# E722: Use common sense: "do not use bare 'except'"
-# E126: Do what feels right
-# E121: Do what feels right
-# E731: "E731 do not assign a lambda expression, use a def" -- don't tell me how to use lambdas
```
Comment (Baekalfen) [inline .flake8]: I kinda liked the snarky comments ;)

File: bootrom/Makefile [hunk 2]
```diff
@@ -3,20 +3,12 @@
 
 all: build
 
-PY := python3
```
Comment (Baekalfen) [inline bootrom/Makefile]: I had these to reflect consistency between the Makefiles. From what I've seen, this is also the norm.

File: bootrom/Makefile [hunk 3]
```diff
@@ -3,20 +3,12 @@
...
-ROOT_DIR := $(shell git rev-parse --show-toplevel)
```
Comment (Baekalfen) [inline bootrom/Makefile]: The purpose was to exactly tell where the output was going, even if the `bootrom` directory moved.

File: bootrom/logo_to_tiles.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+asm_out = ".logo:\n"
+for name in ["P1", "P2", "Y1", "B2", "O", "Y2"]:  # Order is important, so we can't iterate on TILES
+    rows = [int(f"0b{row.replace('.', '0').replace('x', '1')}", base=2) for row in TILES[name].split()]
+    asm_out += f".{name}\n"
+    asm_out += "    DB "
+    line = []
+    for row in rows:
+        line.append(f"${row:02x}")
+    asm_out += ', '.join(line)
+    asm_out += '\n'
```
Comment (Baekalfen) [inline bootrom/logo_to_tiles.py]: Nice rewrite 👍

File: bootrom/Makefile (hunk 3 above)
Comment (krs013) [reply bootrom/Makefile]: That’s fair. I had some ideas about compatibility when I changed it but really there was no reason for me to remove them so it’s good to put it back.

File: .flake8 (hunk 1 above)
Comment (krs013) [reply .flake8]: Haha it was funny at first but I kept losing track of what warning was which... also I’ve wanted to sort them for a while anyway.

File: bootrom/Makefile (hunk 2 above)
Comment (krs013) [reply bootrom/Makefile]: Technically the only python is python3 now ;)

**Comment by krs013** [comment]
So I have a version of this that uses italics but is a little less slanted, but all of these only display the logo for one frame. Not sure why. The scrolling one is cool too and I might like that better, but the logo disappears once it’s done moving on that one too. Once we can fix that I think this is ready to merge.

File: bootrom/logo_to_tiles.py (hunk 4 above)
Comment (krs013) [reply bootrom/logo_to_tiles.py]: Thanks! It was pretty fun. String literals are excellent for this kind of thing.

**Pull request merged**